pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -0,0 +1,136 @@
+package azure
+
+import (
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/Azure/azure-sdk-for-go/arm/containerregistry"
+	azureapi "github.com/Azure/go-autorest/autorest/azure"
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/azure"
+	"k8s.io/kubernetes/pkg/credentialprovider"
+)
+
+var (
+	flagConfigFile = pflag.String("azure-container-registry-config", "",
+		"Path to the file container Azure container registry configuration information.")
+	flagConfigEmail = pflag.String("azure-container-registry-email", "",
```

Comment:
Maybe consider a default value? I don't think ACR uses it at all (currently?) and [even Docker moved away from it](https://github.com/docker/docker/issues/20565). I imagine we'd just put a dummy value here most of the time anyway.